plugins/faustwp/includes/blocks/functions.php

```diff
@@ -1,70 +1,157 @@
 <?php
 /**
- * Functions related to block support.
+ * Block support functions for FaustWP.
  *
  * @package FaustWP
  */
 
 namespace WPE\FaustWP\Blocks;
 
-use function WPE\FaustWP\Utilities\{
-	rrmdir,
-	unzip_to_directory,
-};
+use WP_Error;
 
-if ( ! defined( 'ABSPATH' ) ) {
-	exit;
-}
+require_once ABSPATH . 'wp-admin/includes/file.php';
 
 /**
- * Handle the uploaded blockset file and unzip it.
- * Returns true upon success.
+ * Handles the uploaded blockset file and unzips it.
  *
  * @param array $file The uploaded file details.
- * @return \WP_Error|bool
+ * @return WP_Error|bool
  */
 function handle_uploaded_blockset( $file ) {
-	// Ensure ZipArchive class is available.
-	if ( ! class_exists( 'ZipArchive' ) ) {
-		return new \WP_Error( 'ziparchive_missing', __( 'The ZipArchive class is not available', 'faustwp' ) );
+	global $wp_filesystem;
+	WP_Filesystem();
+
+	$error = validate_uploaded_file( $wp_filesystem, $file );
+	if ( $error ) {
```

Comment:
Shouldn't that always return `error` since in the last line of `validate_uploaded_file` it will return `true`?

Reply:
This should be

	if ( is_wp_error( $error ) ) {

Will fix this and the other.
Thanks for calling this out!